app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorAuthView.tsx

```diff
@@ -0,0 +1,135 @@
+import { FunctionalComponent, FunctionComponent } from 'preact';
```

Comment:
I'm feeling the files to be a bit too cluttered by having many components on each one of them. Can we try to decompose them a bit more?

Reply:
Done. Please verify since it's kind of subjective.